src/ParallelAlgorithms/Events/ObserveNorms.hpp

```diff
@@ -343,6 +384,9 @@ operator()(const typename ObservationValueTag::type& observation_value,
             } else if (tensor_norm_types_[i] == "L2Norm") {
               values.push_back(
                   alg::accumulate(square(components[storage_index]), 0.0));
+            } else if (tensor_norm_types_[i] == "L2IntegralNorm") {
+              values.push_back(definite_integral(
```

Comment:
Can you add an `if (mesh.basis(0) == spectral::basis::FiniteDifferenc) { ERROR(); }` ? The added code won't work for FD, which is actually the reason why I never converted the `ObsereVolumeIntegral` event. I think this is largely okay-ish, as long as there's at least an error and not just some internal Blaze explosion :)

Reply:
Ok! Going forward, can we just add an FD quadrature to `definite_integral` to support integrals on subcell meshes?

Reply:
Yes, that should work I think. The only reason we haven't done anything like that right now is because there's a lot of freedom on how to calculate the integral. If you basically sum up the values at points you get 2nd-order accuracy. This is fine if we're only doing 2nd order FD. If we do higher-order FD then the situation becomes much more complicated. One option is to fit high-order polynomials locally inside each element, but if there's a discontinuity then the fit could be really bad. The technically correct approach would be to do a 3d entire polynomial reconstruction before we observe, then integrate or interpolate that in each FD subcell. Unfortunately that's a pretty major change in the code and also unclear if it's necessary.

Reply:
Ok, sounds like something we'll add once we need it

Reply:
Yep, I or whoever needs it can just add that and generalize it :)